test/Microsoft.AspNetCore.Rewrite.Tests/MiddlewareTests.cs

```diff
@@ -147,6 +147,93 @@ public class MiddlewareTests
+            var options = new RewriteOptions().AddRedirectToWww(statusCode: statusCode);
+            var builder = new WebHostBuilder()
+            .Configure(app =>
+            {
+                app.UseRewriter(options);
+            });
+            var server = new TestServer(builder);
+
+            var response = await server.CreateClient().GetAsync(new Uri("https://example.com"));
+
+            Assert.Equal("https://www.example.com/", response.Headers.Location.OriginalString);
+            Assert.Equal(statusCode, (int)response.StatusCode);
+        }
+
+        [Fact]
+        public async Task CheckRedirectToWww()
+        {
+            var options = new RewriteOptions().AddRedirectToWww();
+            var builder = new WebHostBuilder()
+                .Configure(app =>
+                {
+                    app.UseRewriter(options);
+                });
+            var server = new TestServer(builder);
+
+            var response = await server.CreateClient().GetAsync(new Uri("https://example.com"));
+
+            Assert.Equal("https://www.example.com/", response.Headers.Location.OriginalString);
+            Assert.Equal(StatusCodes.Status307TemporaryRedirect, (int)response.StatusCode);
+        }
+
+        [Fact]
+        public async Task CheckPermanentRedirectToWww()
+        {
+            var options = new RewriteOptions().AddRedirectToWwwPermanent();
+            var builder = new WebHostBuilder()
+                .Configure(app =>
+                {
+                    app.UseRewriter(options);
+                });
+            var server = new TestServer(builder);
+
+            var response = await server.CreateClient().GetAsync(new Uri("https://example.com"));
+
+            Assert.Equal("https://www.example.com/", response.Headers.Location.OriginalString);
+            Assert.Equal(StatusCodes.Status301MovedPermanently, (int)response.StatusCode);
+        }
+
+        [Fact]
+        public async Task CheckNoRedirectToWwwWhenAlreadyWww()
+        {
+            var options = new RewriteOptions().AddRedirectToWww();
+            var builder = new WebHostBuilder()
+            .Configure(app =>
+            {
+                app.UseRewriter(options);
+            });
+            var server = new TestServer(builder);
+
+            var response = await server.CreateClient().GetAsync(new Uri("https://www.example.com"));
+
+            Assert.Null(response.Headers.Location);
+        }
+
+        [Fact]
+        public async Task CheckNoRedirectToWwwWhenLocalhost()
+        {
+            var options = new RewriteOptions().AddRedirectToWww();
+            var builder = new WebHostBuilder()
+            .Configure(app =>
+            {
+                app.UseRewriter(options);
+            });
+            var server = new TestServer(builder);
+
+            var response = await server.CreateClient().GetAsync(new Uri("https://localhost:443"));
+
+            Assert.Null(response.Headers.Location);
+        }
+
```

Comment:
Let's test a few more cases:
- Uri with http
- Uri with many parts, like "https://example.com/test?a=b#foobar" and verify www is added appropriately.
- Uri with just localhost (without port).

You can do this easily by making the test a Theory. See https://github.com/aspnet/BasicMiddleware/blob/dev/test/Microsoft.AspNetCore.HttpsPolicy.Tests/HstsMiddlewareTests.cs#L57